From 767bc49c30679230291a5a9e4c29a1c3906286e6 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 8 Dec 2021 21:37:55 +0200
Subject: [PATCH 12/14] drivers: s32g: pmic: Fix uninitialized usage warning

Initialize function-local variables before setting some of their
members.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-8268
---
 drivers/nxp/s32g/pmic/vr5510.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/nxp/s32g/pmic/vr5510.c b/drivers/nxp/s32g/pmic/vr5510.c
index 7d231ac84..49bc7da65 100644
--- a/drivers/nxp/s32g/pmic/vr5510.c
+++ b/drivers/nxp/s32g/pmic/vr5510.c
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: GPL-2.0+ OR BSD-3-Clause
 /*
- * Copyright 2020 NXP
+ * Copyright 2020-2021 NXP
  */
 #include <common/debug.h>
 #include <endian.h>
@@ -114,7 +114,7 @@ static int vr5510_i2c_write(struct vr5510_inst *dev, uint8_t reg,
 int vr5510_read(struct vr5510_inst *dev, uint8_t reg, uint8_t *buff, int len)
 {
 	unsigned int crc;
-	struct read_msg msg;
+	struct read_msg msg = {.address = 0, .data = 0, .crc = 0};
 
 	if (!valid_register(dev, reg)) {
 		ERROR("Invalid vr5510 register %d\n", reg);
@@ -149,7 +149,7 @@ int vr5510_read(struct vr5510_inst *dev, uint8_t reg, uint8_t *buff, int len)
 int vr5510_write(struct vr5510_inst *dev, uint8_t reg,
 		 const uint8_t *buff, int len)
 {
-	struct read_msg msg;
+	struct read_msg msg = {.address = 0, .data = 0, .crc = 0};
 
 	if (!valid_register(dev, reg)) {
 		ERROR("Invalid vr5510 register %d\n", reg);
-- 
2.17.1

